tests/cuda/CMakeLists.txt

```diff
@@ -14,6 +14,27 @@ include( vecmem-compiler-options-cuda )
 # External dependency/dependencies.
 find_package( CUDAToolkit REQUIRED )
 
+add_library(
+   vecmem_testing_cuda_main
+   SHARED
```

Comment:
This should not have worked... :confused: We'll need to add `-Wl,--no-undefined` for GCC and Clang... (I don't think there's a more elegant way of doing it, than to just hardcode it.)

You see, `cuda_error_handling.cpp` should've been made part of this library if you wanted to use a shared library.

But let's take a further step back. I think this should be made into a static library. Having it as a shared library is just a bit over the top in my mind.

Once you made it into a static library, I would make `cuda_error_handling.cpp` part of that static library. And at the same time remove that source file from the executables.